### PR TITLE
Remove six as a dependency and update dependencies

### PR DESCRIPTION
**Pull request by cosama**
This addresses issue #127.
- [x] Removed six as a dependency and created a small helper function `isstring` for checking if something is a string.
- [x] Add dependencies section to readme
- [x] Change `builtins` imports to `future.builtins`
- [x] Add `pytest-runner` to requirements-dev.txt (required at least on Fedora).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +52,10 @@ with line-by-line highlighting:
 pytest --cov-report html:htmlcov
 ```
 
+## Dependencies
+
+External dependencies are listed in `requirements.txt` and can be installed with `pip` as outlined in the [Installation instructions](#installation-instructions) or manually. The dependencies on `beautifulsoup4`, `lxml` and `html5lib` are necessary for [`pandas`](https://pandas.pydata.org/pandas-docs/stable/install.html#dependencies).
```
Comment (jccurtis) [inline README.md]: @cosama can you also add a note about the `developer` requirements?

File: becquerel/core/utils.py [hunk 2]
```diff
@@ -3,12 +3,25 @@
...
+
+    isinstance("", basestring)
+
+    def isstring(s):
+        """Test for strings in python2 and 3"""
+        return isinstance(s, basestring)
+
+except NameError:
+
+    def isstring(s):
```
Comment (jccurtis) [inline becquerel/core/utils.py]: Can you add comments as to which is for `py2` and `py3`?

File: README.md (hunk 1 above)
Comment (cosama) [reply README.md]: Good idea, let me do that.

File: becquerel/core/utils.py (hunk 2 above)
Comment (cosama) [reply becquerel/core/utils.py]: Yeah good point too. I will do so.

**Comment by cosama** [comment]
I haven't added atom specific tools such as `pyment` to the dependency list. Is that something you would like to add?

**Comment by markbandstra** [comment]
We can add `pyment` in a later PR, not ready for this one.

I restarted the travis build.

**Comment by cosama** [comment]
Thanks @markbandstra . If nobody else has any other comments on this, I think it would be ready to be pulled in.

File: README.md [hunk 3]
```diff
@@ -52,6 +52,10 @@ with line-by-line highlighting:
 pytest --cov-report html:htmlcov
 ```
 
+## Dependencies
+
+External dependencies are listed in `requirements.txt` and can be installed with `pip` as outlined in the [Installation instructions](#installation-instructions) or manually. The dependencies on `beautifulsoup4`, `lxml` and `html5lib` are necessary for [`pandas`](https://pandas.pydata.org/pandas-docs/stable/install.html#dependencies). Developers additionally need [`pytest`](https://docs.pytest.org) and are encouraged to use tools such as [`pylint`](https://pylint.readthedocs.io), [`pycodestyle`](http://pycodestyle.pycqa.org) [`pydocstyle`](http://www.pydocstyle.org) and [`yapf`](https://github.com/google/yapf) for proper code formatting.
```
Comment (markbandstra) [inline README.md]: Does anyone have any opinions on where we break lines in the README and other text files? At 80 characters, every sentence, every paragraph?

Comment (cosama) [reply README.md]: Good point. I would say 80 characters it is as for code, just to be consistent. Let me fix that, also the Copyright notice misses these newlines.

Comment (cosama) [reply README.md]: Some of the links are more than 80 symbols, so I'm actually not sure how to do that. Any thoughts? Just let `[](http://)` excced the line?

Comment (cosama) [reply README.md]: Actually it seems possible to break urls: https://stackoverflow.com/questions/19985235/break-long-lines-in-markdown-code. Let me just break them or is that bad practice when someone wants to actually copy the url from the source code?

Comment (cosama) [reply README.md]: It looks weird. 
```
## Dependencies

External dependencies are listed in `requirements.txt` and can be installed 
with `pip` as outlined in the [Installation instructions
](#installation-instructions) or manually. The dependencies on 
`beautifulsoup4`, `lxml` and `html5lib` are necessary for [`pandas`
](https://pandas.pydata.org/pandas-docs/stable/install.html#dependencies).
Developers additionally need [`pytest`](https://docs.pytest.org) and are 
encouraged to use tools such as [`pylint`](https://pylint.readthedocs.io),
[`pycodestyle`](http://pycodestyle.pycqa.org), [`pydocstyle`](
http://www.pydocstyle.org) and [`yapf`](https://github.com/google/yapf)
for proper code formatting.
```

Maybe just add line breaks every sentence or paragraph would make more sense. Any thoughts?

Comment (jccurtis) [reply README.md]: We could use markdown [references](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links):

```markdown
## Dependencies

External dependencies are listed in `requirements.txt` and can be installed
with `pip` as outlined in the  [Installation instructions][0] or manually.
The dependencies on `beautifulsoup4`, `lxml` and `html5lib` are necessary for
[pandas][1]. Developers additionally need [`pytest`][2] and are encouraged to
use tools such as [`pylint`][3], [`pycodestyle`][4], [`pydocstyle`][5] and
[`yapf`][6] for proper code formatting.

[0]: #installation-instructions
[1]: https://pandas.pydata.org/pandas-docs/stable/install.html#dependencies
[2]: https://docs.pytest.org
[3]: https://pylint.readthedocs.io
[4]: http://pycodestyle.pycqa.org
[5]: http://www.pydocstyle.org
[6]: https://github.com/google/yapf
```

Which renders on github with the soft breaks being registered as line breaks [which shouldn't be possible without extra spaces at the end of the line...](https://gist.github.com/shaunlebron/746476e6e7a4d698b373) :shipit:

**Maybe this doesn't happen in the markdown rendering for `md` files**

## Dependencies

External dependencies are listed in `requirements.txt` and can be installed 
with `pip` as outlined in the  [Installation instructions][0] or manually.
The dependencies on `beautifulsoup4`, `lxml` and `html5lib` are necessary for
[pandas][1]. Developers additionally need [`pytest`][2] and are encouraged to
use tools such as [`pylint`][3], [`pycodestyle`][4], [`pydocstyle`][5] and
[`yapf`][6] for proper code formatting.

[0]: #installation-instructions
[1]: https://pandas.pydata.org/pandas-docs/stable/install.html#dependencies
[2]: https://docs.pytest.org
[3]: https://pylint.readthedocs.io
[4]: http://pycodestyle.pycqa.org
[5]: http://www.pydocstyle.org
[6]: https://github.com/google/yapf

Comment (cosama) [reply README.md]: How the hell did you manage to render that? Looks good.

File: README.md [hunk 4]
```diff
@@ -52,6 +52,23 @@ with line-by-line highlighting:
...
+[`pylint`][3], [`pycodestyle`][4], [`pydocstyle`][5] and [`yapf`][6] for
+proper code formatting.
+
+[0]: #installation-instructions
+[1]: https://pandas.pydata.org/pandas-docs/stable/install.html#dependencies
+[2]: https://docs.pytest.org
+[3]: https://pylint.readthedocs.io
+[4]: http://pycodestyle.pycqa.org
+[5]: http://www.pydocstyle.org
+[6]: https://github.com/google/yapf
```
Comment (jccurtis) [inline README.md]: Nice one @cosama I'm glad this worked out

File: requirements-dev.txt [hunk 5]
```diff
@@ -1,3 +1,4 @@
 -r requirements.txt
 pytest>=3.0.0
 pytest-cov
+pytest-runner
```
Comment (jccurtis) [inline requirements-dev.txt]: Why was this added?

File: becquerel/core/utils.py (hunk 2 above)
Comment (jccurtis) [reply becquerel/core/utils.py]: @cosama just a reminder about this

Comment (cosama) [reply becquerel/core/utils.py]: The version is mentioned in the docstring now.

File: becquerel/core/energycal.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 """"Energy calibration classes"""
 
 from abc import ABCMeta, abstractmethod, abstractproperty
-from builtins import dict, super, zip  # pylint: disable=redefined-builtin
+from future.builtins import dict, super, zip
```
Comment (jccurtis) [inline becquerel/core/energycal.py]: Are we planning to add this `pylint` disable to the `pylintrc`

Comment (cosama) [reply becquerel/core/energycal.py]: They have been there for a while. However, pylint only doesn't like the import if it comes from `builtins`. At least on my system it did not complain about those includes from `future.builtins`, thus I removed them.
